velox/vector/ComplexVector.h

```diff
@@ -512,15 +518,17 @@ class MapVector : public BaseVector {
   }
 
   BufferPtr mutableOffsets(size_t size) {
-    if (offsets_ && offsets_->capacity() >= size * sizeof(vector_size_t)) {
+    if (offsets_ && offsets_->isMutable() &&
```

Comment:
nit: maybe these checks can be moved to some common function which can be inlined..